src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -273,18 +273,111 @@ public void Shuffle<T>(T[] values)
         /// </remarks>
         public void Shuffle<T>(Span<T> values)
         {
-            int n = values.Length;
+            var chooser = new IncreasingUniform(this);
+            for (int i = 1; i < values.Length; i++)
+            {
+                int index = chooser.NextIndex();
+                (values[i], values[index]) = (values[index], values[i]);
+            }
+        }
+
+        private ref struct IncreasingUniform
```

Comment:
It doesn't have to be a ref struct.

```suggestion
        private struct IncreasingUniform
```